### DISCUSSION TITLE
Custom Script new Device, Problem with Module Bays

### DISCUSSION THREAD

**Discussion by heidl89** [netbox-community/netbox#17842] [General]
Hi. 
I've made a custom script, which creates a new device with some input data (device-types, gps, ...)  and api calls for other systems.... 
We created for every model a device-type. 

First i am creating the device, then interface and ip. 
That works fine. 

My Problem is, the last save() command. 
I need this for updating the device with the setattr
But then my module bays and power outlets are not visible in the gui. 

Here's my code: 

    def run (self, data, commit):

        if not data['gps_lat']:   
                data['gps_lat'] = 0
        if not data['gps_long']:
                data['gps_long'] = 0

        
        
        device = Device(
                name = data['device_name'],
                device_type = data['device_type'],
                role = data['device_role'],
                status = data['status'],
                latitude = data['gps_lat'],
                longitude = data['gps_long'],
                site_id = data['site'].id
         )
         
        if data['location']:             
                device.location = data['location']
        
        device.full_clean()
        device.save()

        
        interface = Interface(
                device = device,
                name = "vmgmt0",
                type = InterfaceTypeChoices.TYPE_VIRTUAL,
                enabled = True,
                mgmt_only = True
        )

        interface.full_clean()
        interface.save()
        
        ip = IPAddress(
            address = data['ipaddress'],
            status = IPAddressStatusChoices.STATUS_ACTIVE,
            assigned_object = interface
            
        )
        ip.full_clean()
        ip.save()
        
        setattr(device,"primary_ip4", ip)

        device.save()
             
        self.log_success(f"Created new Device {device}")

Greets Markus

**Comment by heidl89** [comment]
**Update**:
 I just investigated this Problem and found out, that the bays other components were created as you can see in the change-log.

My Problem is that they still aren't visible. :(


![nbx_change_log](https://github.com/user-attachments/assets/03018e14-376a-4f68-aabf-c31b0f0b55cc)

**Reply by candlerb** [reply]
> My Problem is that they still aren't visible. :(

Define what you mean by "aren't visible". Do you mean that at `/dcim/devices/N/` the tab for power ports is not present?  But if you go to `/dcim/devices/N/power-ports/` (by editing the URL) are the power ports visible?

See if this fixes the problem for devices you've just created:

```
/opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py calculate_cached_counts
```

If it does, then the trouble is that the tabs are hidden when the count is zero, and the counts aren't getting updated when the device is created.

It would be a good idea if you could give basic information about your system, such as what Netbox version you're running.

**Comment by heidl89** [comment]
Duplicate: [#17861 ](https://github.com/netbox-community/netbox/discussions/17861)
